.goreleaser.yaml

```diff
@@ -5,24 +5,81 @@ before:
     - go mod tidy
 
 universal_binaries:
-  - replace: true
+  - id: oasis-darwin-universal
+    ids:
+      - oasis-darwin-amd64
+      - oasis-darwin-arm64
+    replace: true
     name_template: oasis
 
+env:
+  - CGO_ENABLED=1
+
 builds:
-  - binary: oasis
+  - id: oasis-linux-amd64
+    binary: oasis
+    env:
+      - CC=x86_64-linux-gnu-gcc
+      - CXX=x86_64-linux-gnu-g++
+    goos:
+      - linux
+    goarch:
+      - amd64
+    flags:
+      - -trimpath
+    ldflags:
+      - -buildid=
+      - "{{.Env.GOLDFLAGS_VERSION}}"
+  - id: oasis-linux-arm64
+    binary: oasis
+    env:
+      - CC=aarch64-linux-gnu-gcc
+      - CXX=aarch64-linux-gnu-g++
+    goos:
+      - linux
+    goarch:
+      - arm64
+    flags:
+      - -trimpath
+    ldflags:
+      - -buildid=
+      - "{{.Env.GOLDFLAGS_VERSION}}"
+  - id: oasis-darwin-amd64
+    binary: oasis
+    env:
+      - CC=o64-clang
+      - CXX=o64-clang++
+    goos:
+      - darwin
+    goarch:
+      - amd64
+    flags:
+      - -trimpath
+    ldflags:
+      - -buildid=
+      - "{{.Env.GOLDFLAGS_VERSION}}"
+  - id: oasis-darwin-arm64
+    binary: oasis
+    env:
+      - CC=oa64-clang
+      - CXX=oa64-clang++
+    goos:
+      - darwin
+    goarch:
+      - arm64
```

Comment:
There is quite some duplicate code here. Did you consider using [github actions matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs)?

Reply:
This should be a bit better now, I used yaml's map merge.

Reply:
I suppose moving `build-common` outside of the `builds` doesn't work? Sth like:

```yaml
x-build-common: &build-common
  flags:
    - -trimpath
  ldflags:
    - -buildid=
    - "{{.Env.GOLDFLAGS_VERSION}}"
```

Reply:
It doesn't, unfortunately. I didn't find a way to declare a map without it also being part of the parsed result, i.e. with your example, goreleaser's struct would need a field called XBuildCommon or some such. You can only put anchors on actual data and then reference them.